RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -971,10 +974,307 @@ vector<TransientVertex> DAClusterizerInZ_vect::vertices(const vector<reco::Trans
   return clusters;
 }
 
-vector<vector<reco::TransientTrack> > DAClusterizerInZ_vect::clusterize(
+vector<TransientVertex> DAClusterizerInZ_vect::vertices_in_blocks(const vector<reco::TransientTrack>& tracks) const {
+  vector<reco::TransientTrack> sorted_tracks;
+  vector<pair<float, float>> vertices_tot;  // z, rho for each vertex
+  for (unsigned int i = 0; i < tracks.size(); i++) {
+    sorted_tracks.push_back(tracks[i]);
+  }
+  double rho0, beta;
+  std::sort(sorted_tracks.begin(),
+            sorted_tracks.end(),
+            [](const reco::TransientTrack& a, const reco::TransientTrack& b) -> bool {
+              return (a.stateAtBeamLine().trackStateAtPCA()).position().z() <
+                     (b.stateAtBeamLine().trackStateAtPCA()).position().z();
+            });
+
+  unsigned int nBlocks = (unsigned int)std::floor(sorted_tracks.size() / (block_size_ * (1 - overlap_frac_)));
```

Comment:
I'm curious, if nBlocks is 1 (e.g. by virtue of setting block size to max int, does the rest of implementation become `DAClusterizerInZ_vect::vertices` ?
If yes, can we replace/remove the non-block impl?

Reply:
In principle yes it goes back to "non-block" implementation.

Reply:
Can `nBlocks` ever become 0 here? In that case, `beta` would remain unassigned, as pointed out by the static analyzer